docs/Evaluation.md

```diff
@@ -1,18 +1,86 @@
+
+There is an additional difference between ecological object detection like tree crowns and traditional computer vision methods. Instead of a single or set of easy to differentiate ground truths, we could have 60 or 70 objects that overlap in an image. How to best assign each prediction to each ground truth? DeepForest uses the [hungarian matching algorithm](https://thinkautonomous.medium.com/computer-vision-for-tracking-8220759eee85) to assign predictions to ground truth based on maximum IoU overlap. This is slow compared to the methods above, and so isn't a good choice for running hundreds of times during model training see config["validation"]["val_accuracy_interval"] for setting the frequency of the evaluate callback for this metric.
+
+# Calculating Evaluation Metrics 
+
+## Torchmetrics and loss scores
+
+These metrics are largely used during training to keep track of model performance. They are relatively fast and will be automatically run during training. 
+
+```    
+    m = main.deepforest()
+    csv_file = get_data("OSBS_029.csv")
+    root_dir = os.path.dirname(csv_file)
+    m.config["validation"]["csv_file"] = csv_file
+    m.config["validation"]["root_dir] = root_dir
+    results = m.trainer.validate(m)
+```
+This creates a dictionary of the average IoU ('iou') as well as 'iou' for each class. Here there is just one class, 'Tree'. Then the COCO mAP scores. See Further Reading above for explanation of mAP level scores. The val_bbox_regression is the loss function of the object detection box head, the loss_classification is the loss function of the object classification head.
+```
+        classes                    0.0
+           iou              0.6305446564807566
+        iou/cl_0            0.6305446564807566
+           map              0.04219449311494827
+         map_50             0.11141198128461838
+         map_75            0.025357535108923912
+        map_large                  -1.0
+       map_medium           0.05097917467355728
+      map_per_class                -1.0
+        map_small                   0.0
+          mar_1            0.008860759437084198
+         mar_10             0.03417721390724182
+         mar_100            0.08481013029813766
+    mar_100_per_class              -1.0
+        mar_large                  -1.0
+       mar_medium           0.09436620026826859
+        mar_small                   0.0
+   val_bbox_regression      0.5196723341941833
+   val_classification       0.4998389184474945
+```
+
+> **_Advanced tip:_**  Users can set the frequency of pytorch lightning evaluation using kwargs passed to main.deepforest.create_trainer(). For example [check_val_every_n_epochs](https://lightning.ai/docs/pytorch/stable/common/trainer.html#check-val-every-n-epoch).
+
+## Recall and Precision at a fixed IoU Score
+
+To get a recall and precision at a set IoU evaluation score, specify an annotations file an using the m.evaluate method.
```

Comment:
To get a recall and precision at a set IoU evaluation score, specify an annotations' file using the m.evaluate method.